### DISCUSSION TITLE
Rest Query Values Lose Trailing Zero in Line Chart

### DISCUSSION THREAD

**Discussion by mmays77** [Budibase/budibase#12766] [Help] [answered]
I have a rest query that is returning currency values that are properly formatted json as follows (even though the text editor has squashed the indentation):

{
  "data": [
    {
      "DividendMonth": "2023-11",
      "~Joint-Tenant~": "180.47",
      "~Mark IRA~": "74.30",
    },
    {
      "DividendMonth": "2023-12",
      "~Joint-Tenant~": "810.70",
      "~Mark IRA~": "520.24",
    },
    {
      "DividendMonth": "2024-01",
      "~Joint-Tenant~": "373.80",
      "~Mark IRA~": "237.80",
    }
  ]
}

These json strings are loaded into a rest query in Budibase. In the Budibase rest query section, the raw data look good, the schema looks good. When the rest query is used to load a line chart, however, the values that end in zero are truncated, so they show up as "810.7" instead of "810.70".

This happens whether the data values in the rest query schema are set to "text" or "number" data type. There is no "currency" data type, and I cannot figure out a way to force the formatting of the 2 digit decimal to keep the line chart from dropping the trailing zero. 

What am I missing? Thanks in advance if you can help.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @mmays77,

Firstly, you need to remove the last comma off each field inside those objects. JSON does not allow trailing commas in arrays or objects.

**Json Data**
```json
{
  "data": [
    {
      "DividendMonth": "2023-11",
      "Joint-Tenant": "180.47",
      "Mark IRA": "74.30"
    },
    {
      "DividendMonth": "2023-12",
      "Joint-Tenant": "810.70",
      "Mark IRA": "520.24"
    },
    {
      "DividendMonth": "2024-01",
      "Joint-Tenant": "373.80",
      "Mark IRA": "237.80"
    }
  ]
}
```

I can see the issue you are describing, its likely related to how the chart components interact with numbers, a core fix is needed to be made here in order to display the numbers as you've described. I've included a link to the created issue [here](https://github.com/Budibase/budibase/issues/12768).
